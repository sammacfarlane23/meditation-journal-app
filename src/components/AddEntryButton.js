import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus } from '@fortawesome/free-solid-svg-icons';
import EntryModal from './EntryModal';
import { startCreateEntry } from '../actions/entries';

export default () => {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const dispatch = useDispatch();

  const showModal = () => {
    setIsModalOpen(true);
  };

  const onSubmit = (entry) => {
    dispatch(startCreateEntry(entry));
    setIsModalOpen(false);
  };

  const closeModal = () => {
    setIsModalOpen(false);
  };

  return (
    <div className='button-position'>
      {!isModalOpen && (
        <button className='button--big circular-container' onClick={showModal}>
          <FontAwesomeIcon icon={faPlus} size='2x' />
        </button>
      )}
      <EntryModal
        showModal={isModalOpen}
        onSubmit={onSubmit}
        closeModal={closeModal}
      />
    </div>
  );
};
