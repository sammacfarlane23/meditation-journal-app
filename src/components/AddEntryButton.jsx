import React from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus } from '@fortawesome/free-solid-svg-icons';
import EntryModal from './EntryModal';
import { startCreateEntry } from '../actions/entries';

export class AddEntryButton extends React.Component {
  state = { show: false };

  showModal = () => {
    this.setState(() => ({ show: true }));
  };

  onSubmit = (entry) => {
    this.props.startCreateEntry(entry);
    this.setState({ show: false });
  };

  closeModal = () => {
    this.setState({ show: false });
  };
  render() {
    return (
      <div className='button-position'>
        {!this.state.show && (
          <button
            className='button--big circular-container'
            onClick={this.showModal}
          >
            <FontAwesomeIcon icon={faPlus} size='2x' />
          </button>
        )}
        <EntryModal
          showModal={this.state.show}
          onSubmit={this.onSubmit}
          closeModal={this.closeModal}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startCreateEntry: (entry) => dispatch(startCreateEntry(entry)),
});

export default connect(undefined, mapDispatchToProps)(AddEntryButton);
