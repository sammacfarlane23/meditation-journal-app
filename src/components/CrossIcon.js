import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

export default (props) => (
  <div>
    <button className='footer-button' onClick={props.showSearch}>
      <FontAwesomeIcon icon={faTimes} className='white-button' size='2x' />
    </button>
  </div>
);
