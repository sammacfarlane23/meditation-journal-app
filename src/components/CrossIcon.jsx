import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

export default (props) => (
  <div>
    <button className='button--big' onClick={props.showSearch}>
      <FontAwesomeIcon icon={faTimes} size='2x' />
    </button>
  </div>
);
