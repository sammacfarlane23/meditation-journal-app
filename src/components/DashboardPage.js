import React from 'react';
import EntryList from './EntryList';
import Footer from './Footer';

export default () => (
  <div className='dashboard-layout'>
    <div className='content-container'>
      <EntryList />
    </div>
    <Footer />
  </div>
);
