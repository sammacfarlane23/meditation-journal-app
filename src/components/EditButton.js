import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEdit } from '@fortawesome/free-solid-svg-icons';

export default () => (
  <div>
    <FontAwesomeIcon icon={faEdit} className='edit-icon' />
  </div>
);
