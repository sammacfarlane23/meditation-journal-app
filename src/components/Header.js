import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import SearchIcon from './SearchIcon';
import EntryListFilters from './EntryListFilters';
import { startLogout } from '../actions/auth';
import { setTextFilter } from '../actions/filters';
import LogoutIcon from './LogoutIcon';
import CrossIcon from './CrossIcon';

export class Header extends React.Component {
  state = {
    showSearchBar: false,
  };

  showHideSearch = () => {
    this.props.setTextFilter('');
    this.setState(() => ({
      showSearchBar: !this.state.showSearchBar,
    }));
  };

  componentDidMount() {
    document.addEventListener('click', this.handleClickOutside, true);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.handleClickOutside, true);
  }

  handleClickOutside = (e) => {
    const domNode = ReactDOM.findDOMNode(this);

    if (!domNode || !domNode.contains(e.target)) {
      this.props.setTextFilter('');
      this.setState({ showSearchBar: false });
    }
  };

  render() {
    return (
      <header className='header'>
        <div className='content-container'>
          <div className='header__content'>
            {!this.state.showSearchBar && (
              <Link className='header__title' to='/dashboard'>
                <h1>JOURNAL</h1>
              </Link>
            )}
            {this.state.showSearchBar && (
              <EntryListFilters tabIndex='0' onBlur={this.showHideSearch} />
            )}
            <div className='right-header'>
              {this.state.showSearchBar ? (
                <CrossIcon showSearch={this.showHideSearch} />
              ) : (
                <SearchIcon showSearch={this.showHideSearch} />
              )}
              <LogoutIcon startLogout={this.props.startLogout} />
            </div>
          </div>
        </div>
      </header>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  startLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
