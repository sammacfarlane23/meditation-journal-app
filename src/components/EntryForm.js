import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import EditButton from './EditButton';

export default class EntryForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      entryText: props.entry ? props.entry.entryText : '',
      createdAt: props.entry ? moment(props.entry.createdAt) : moment(),
      error: '',
      calendarFocused: false,
      title: props.entry ? props.entry.title : '',
      editDate: false,
    };
  }

  onEntryTextChange = (e) => {
    const entryText = e.target.value;
    this.setState(() => ({ entryText }));
  };

  onRemove = () => {
    this.props.onRemove(this.props.entry);
  };

  onTitleChange = (e) => {
    const title = e.target.value;
    this.setState(() => ({ title }));
  };

  onTimeChange = (e) => {
    const currentDate = moment(this.state.createdAt).startOf('day');
    const timeOfDay = moment.duration(
      moment(e.target.value, 'HH:mm') -
        moment(e.target.value, 'HH:mm').startOf('day')
    );
    const createdAt = moment(currentDate).add(timeOfDay);
    this.setState(() => ({ createdAt }));
  };

  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };

  onDateChange = (createdAt) => {
    const timeOfDay =
      moment(this.state.createdAt) -
      moment(this.state.createdAt).startOf('day');
    const addTime = moment(moment(createdAt).startOf('day') + timeOfDay);
    if (createdAt) {
      this.setState(() => ({ createdAt: addTime }));
    }
  };

  onEditDate = () => {
    this.setState(() => ({ editDate: true }));
  };

  closeModal = () => {
    this.props.closeModal();
  };

  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.entryText) {
      this.setState(() => ({
        error: 'Please add an entry',
      }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        entryText: this.state.entryText,
        createdAt: this.state.createdAt.valueOf(),
        title: this.state.title,
      });
    }
  };

  render() {
    return (
      <form className='form' onSubmit={this.onSubmit}>
        <input
          className='title'
          type='text'
          placeholder='Title (optional)'
          value={this.state.title ? this.state.title : ''}
          onChange={this.onTitleChange}
        />
        <hr className='horizontal-line' />
        {this.state.error && <p className='form__error'>{this.state.error}</p>}
        <textarea
          rows='10'
          value={this.state.entryText}
          className='text-area'
          placeholder='Describe your experience here...'
          onChange={this.onEntryTextChange}
        ></textarea>
        {!this.state.editDate && (
          <div className='createdAt-section'>
            <p className='createdAt-section__text'>
              {moment(this.state.createdAt).format('DD MMM YYYY HH:mm')}
            </p>
            <button
              type='button'
              className='edit-button'
              onClick={this.onEditDate}
            >
              <EditButton />
            </button>
          </div>
        )}
        {this.state.editDate && (
          <div className='date-edit'>
            <div className='single-date-picker'>
              <SingleDatePicker
                date={this.state.createdAt}
                onDateChange={this.onDateChange}
                onFocusChange={this.onFocusChange}
                focused={this.state.calendarFocused}
                numberOfMonths={1}
                isOutsideRange={() => false}
                displayFormat='DD/MM/YYYY'
                isOutsideRange={(d) => d.isAfter(moment())}
                withPortal={true}
              />
            </div>
            <input
              className='input-time'
              type='time'
              onChange={this.onTimeChange}
              value={moment(this.state.createdAt).format('HH:mm')}
              max={
                moment()
                  .startOf('day')
                  .isSame(moment(this.state.createdAt).startOf('day'))
                  ? moment().format('HH:mm')
                  : '23:59'
              }
            />
          </div>
        )}
        <div className='button-container'>
          <div className='remove-button-container'>
            {this.props.canEdit && (
              <button
                type='button'
                className='button button--delete'
                onClick={this.onRemove}
              >
                Delete
              </button>
            )}
          </div>
          <div className='other-button-container'>
            <button className='button button--right'>Save</button>
            <button
              type='button'
              onClick={this.closeModal}
              className='button button--right button--cancel'
            >
              Cancel
            </button>
          </div>
        </div>
      </form>
    );
  }
}
