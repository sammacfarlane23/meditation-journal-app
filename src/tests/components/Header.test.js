import React from 'react';
import { Header } from '../../components/Header';
import { render } from '@testing-library/react';

test('should render Header correctly', () => {
  const component = render(<Header />);
  expect(component).toMatchSnapshot();
});
