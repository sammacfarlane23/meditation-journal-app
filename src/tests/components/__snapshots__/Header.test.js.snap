// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <h1
        className="header__title"
      >
        JOURNAL
      </h1>
      <div
        className="header__right"
      >
        <Component
          showSearch={[Function]}
        />
        <Component />
      </div>
    </div>
  </div>
</header>
`;
